File: app/src/main/java/com/arny/flightlogbook/data/db/FlightTypeValueDAO.java
package com.arny.flightlogbook.data.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.arny.flightlogbook.data.models.FlightType;
import com.arny.flightlogbook.data.models.FlightTypeValue;

import java.util.List;

@Dao
public interface FlightTypeValueDAO {
	@Query("SELECT * FROM flight_type_values")
	List<FlightTypeValue> queryFlightTypeValues();

	@Query("SELECT * FROM flight_type_values WHERE _id=:id")
	FlightTypeValue queryFlightTypeValue(long id);

	@Query("SELECT * FROM flight_type_values WHERE type=:id")
	List<FlightTypeValue> queryFlightTypeValuesByType(long id);

	@Query("DELETE FROM flight_type_values WHERE _id=:id")
	int delete(long id);

	@Query("DELETE FROM flight_type_values")
	int delete();

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	long insert(FlightTypeValue flightTypeValue);
}

File: app/src/main/java/com/arny/flightlogbook/data/db/daos/FlightTypeValueDAO.kt
package com.arny.flightlogbook.data.db.daos

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import com.arny.flightlogbook.data.models.FlightTypeValue

@Dao
interface FlightTypeValueDAO : BaseDao<FlightTypeValue> {
    @Query("SELECT * FROM flight_type_values")
    fun queryFlightTypeValues(): List<FlightTypeValue>

    @Query("SELECT * FROM flight_type_values WHERE _id=:id")
    fun queryFlightTypeValue(id: Long): FlightTypeValue

    @Query("SELECT * FROM flight_type_values WHERE type=:id")
    fun queryFlightTypeValuesByType(id: Long): List<FlightTypeValue>

    @Query("DELETE FROM flight_type_values WHERE _id=:id")
    fun delete(id: Long): Int

    @Query("DELETE FROM flight_type_values")
    fun delete(): Int
}
